core/controllers/profile_test.py

```diff
@@ -45,6 +45,31 @@ def test_going_somewhere_else_while_signing_in_logs_user_out(self):
 
         self.logout()
 
+    def test_to_check_url_redirection_in_signup(self):
+        """To validate the redirections from return_url."""
+        self.login(self.EDITOR_EMAIL)
+        response = self.get_html_response(feconf.SIGNUP_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        # Registering this user fully.
+        self.post_json(
+            feconf.SIGNUP_DATA_URL,
+            {'username': 'abc', 'agreed_to_terms': True},
+            csrf_token=csrf_token)
+
+        response = self.get_html_response(
+            '/signup?return_url=https://google.com', expected_status_int=302)
+        response = self.get_html_response(
+            '/signup?return_url=//google.com', expected_status_int=302)
+        response = self.get_html_response(
+            '/signup?return_url=/page', expected_status_int=302)
+
+        self.assertIn('/', response.headers['location'])
```

Comment:
I believe this is only checking it for the most recent response. You need to do the assertion immediately after you get the response in each case.

Also, check equality for the whole URL, don't just check containment. That would probably have caught the issue.

Reply:
Sorry Missed it !

Also, you mean to use assertEqual(‘https://localhost:8081’,response.headers[‘location’]) ?

I dint check the equality of whole url thinking in mind that other devs may use other hosts or ports which can cause issue.

Reply:
You can strip the host/port part out, and check equality of the remaining string.

Reply:
Adding Third party libraries like urlparse for the same or regex only?
parsing such with urlparse will be very smooth but with urlparse we may find difficulties while migrating to python3.

Reply:
No need to import third-party libraries.

Reply:
sure